app/models/category.rb

```diff
@@ -7,4 +7,5 @@ class Category < ApplicationRecord
   belongs_to :parent, class_name: Category.name, optional: true
   has_many :products, dependent: :destroy
   scope :parent_categories, ->{where parent_id: nil}
+  scope :children_of, ->(id){where parent_id: id}
```

Comment:
Sao em ko dùng association nhỉ
`has_many :children, class_name: Category.name, foreign_key: :parent_id`
thì ở đây chỉ cần lấy category.children là lấy dc toàn bộ category con của nó

Reply:
dạ do model này có association children rồi nên ở commit sau em thay scope này bằng `find_by` rồi gọi nó luôn ạ